libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt

```diff
@@ -31,8 +31,16 @@ import com.mapbox.navigation.ui.shield.model.RouteShield
  * removing the listener with [MapboxCarMap.unregisterObserver].
  */
 @OptIn(MapboxExperimental::class)
-class RoadLabelSurfaceLayer(
-    val carContext: CarContext,
+class RoadLabelSurfaceLayer internal constructor(
+    delegate: RoadLabelSurfaceLayerDelegate,
+) : MapboxCarMapObserver by delegate {
+
+    constructor(carContext: CarContext) : this(RoadLabelSurfaceLayerDelegate(carContext))
```

Comment:
the carContext is in MapboxCarMap (get it from the onAttached), doesn't have to be this change but we'll want to remove it from the constructor